--- Makefile	2007-05-26 11:52:33.000000000 +0200
+++ dynamips-0.2.7/Makefile	2007-09-11 21:15:40.508475542 +0200
@@ -32,7 +32,7 @@
 MIPS64_ARCH_INC_FILE=\"mips64_$(DYNAMIPS_ARCH)_trans.h\"
 PPC32_ARCH_INC_FILE=\"ppc32_$(DYNAMIPS_ARCH)_trans.h\"
 
-CFLAGS+=-g -Wall -O3 -fomit-frame-pointer \
+CFLAGS+= $(RPM_CFLAGS) -fomit-frame-pointer \
 	-DJIT_ARCH=\"$(DYNAMIPS_ARCH)\" -DJIT_CPU=CPU_$(DYNAMIPS_ARCH) \
 	-DMIPS64_ARCH_INC_FILE=$(MIPS64_ARCH_INC_FILE) \
 	-DPPC32_ARCH_INC_FILE=$(PPC32_ARCH_INC_FILE) \
@@ -51,9 +51,9 @@
 else
 ifeq ($(shell uname), Linux)
    PTHREAD_LIBS?=-lpthread
-#   PCAP_LIB=-lpcap
-   CFLAGS+=-I/usr/include -I. $(PTHREAD_CFLAGS)
-   LIBS=-L/usr/lib -L. /usr/lib/libelf.a $(PTHREAD_LIBS)
+   PCAP_LIB=-lpcap
+   CFLAGS+=-I. $(PTHREAD_CFLAGS)
+   LIBS=-L. -lelf $(PTHREAD_LIBS)
    DESTDIR=/usr
 else
 ifeq ($(shell uname -s), Darwin)
@@ -201,15 +201,15 @@
 $(PROG): mips64_microcode_dump.inc ppc32_microcode_dump.inc \
 	$(LEX_C) $(C_OBJS) $(A_OBJS)
 	@echo "Linking $@"
-	@$(CC) -o $@ $(C_OBJS) $(A_OBJS) $(LIBS)
+	$(CC) -o $@ $(C_OBJS) $(A_OBJS) $(LIBS)
 
 udp_send$(BIN_EXT): udp_send.c net.c
 	@echo "Linking $@"
-	@$(CC) -Wall $(CFLAGS) -o $@ udp_send.c net.c $(LIBS)
+	$(CC) -Wall $(CFLAGS) -o $@ udp_send.c net.c $(LIBS)
 
 rom2c$(BIN_EXT): rom2c.c
 	@echo "Linking $@"
-	@$(CC) -Wall $(CFLAGS) -o $@ rom2c.c $(LIBS)
+	$(CC) -Wall $(CFLAGS) -o $@ rom2c.c $(LIBS)
 
 mips64_microcode_dump.inc: rom2c$(BIN_EXT) mips64_microcode
 	@./rom2c mips64_microcode mips64_microcode_dump.inc 0xbfc00000
@@ -219,7 +219,7 @@
 
 asmdefs$(BIN_EXT): asmdefs.c mips64.h
 	@echo "Linking $@"
-	@$(CC) -Wall $(CFLAGS) -o $@ asmdefs.c
+	$(CC) -Wall $(CFLAGS) -o $@ asmdefs.c
 
 asmdefs.h: asmdefs$(BIN_EXT)
 	@echo "Building assembly definitions header file"
@@ -227,7 +227,7 @@
 
 nvram_export$(BIN_EXT): nvram_export.c
 	@echo "Linking $@"
-	@$(CC) -Wall $(CFLAGS) -o $@ nvram_export.c
+	$(CC) -Wall $(CFLAGS) -o $@ nvram_export.c
 
 install: $(PROG) nvram_export
 	@echo "Installing"
@@ -261,11 +261,11 @@
 
 .S.o:
 	@echo "Assembling $<"
-	@$(CC) $(CFLAGS) $(INCLUDE) -c -o $*.o $<
+	$(CC) $(CFLAGS) $(INCLUDE) -c -o $*.o $<
 
 .c.o:
 	@echo "Compiling $<"
-	@$(CC) $(CFLAGS) $(INCLUDE) -c -o $*.o $<
+	$(CC) $(CFLAGS) $(INCLUDE) -c -o $*.o $<
 
 .l.c:
 	$(LEX) -o$*.c $<
